service/storage_proxy.cc

```diff
@@ -1456,6 +1456,10 @@ storage_proxy::response_id_type storage_proxy::create_write_response_handler(key
     shared_ptr<abstract_write_response_handler> h;
     auto& rs = ks.get_replication_strategy();
 
+    if (tr_state.has_opentelemetry()) {
+        tr_state.get_opentelemetry_ptr()->set_replicas(targets);
+    }
```

Comment:
Maybe similarly to other methods in `trace_state.hh` make a helper method for `set_replicas`?

(see `set_username` in `trace_state.hh`)
```cpp
inline void set_username(const trace_state_ptr& p, const std::optional<auth::authenticated_user>& user) {
    if (p.has_tracing()) {
        p.get_tracing_ptr()->set_username(user);
    }
}
```